### PR TITLE
[BPK-2526] Implemented BpkLinkSpan, set links to bpkPrimary color for…

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkLinkSpan.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import android.content.Context
+import android.text.TextPaint
+import android.text.style.ClickableSpan
+import android.view.View
+import net.skyscanner.backpack.util.BpkTheme
+
+class BpkLinkSpan<T>(
+  context: Context,
+  private val link: T,
+  private val linksHandler: (T) -> Unit
```
Comment (tiagoengel) [inline Backpack/src/main/java/net/skyscanner/backpack/text/BpkLinkSpan.kt]: nit: shouldn't this be called `linkHandler` (singular) as it refers to only one link?

File: UNRELEASED.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 > Place your changes below this line.
 
+**Added**:
+
+- `BpkLinkSpan` span.
+- `BpkText` uses `bpkPrimary` color for the links.
```
Comment (tiagoengel) [inline UNRELEASED.md]:
```suggestion
- `BpkText` now uses `bpkPrimary` colour for links. 
```

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt [hunk 3]
```diff
@@ -76,6 +77,7 @@ open class BpkText @JvmOverloads constructor(
     }
 
   private var _textColour: ColorStateList? = null
+  private var _textColourLink: ColorStateList? = null
```
Comment (tiagoengel) [inline Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt]: nit: this should be `color` not `colour`. We used American english in code.

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkLinksSpanTests.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  private val textView = TextView(testContext).apply { setBackgroundColor(Color.WHITE) }
+
+  @Before
+  fun setup() {
+    setDimensions(20, 50)
+  }
+
+  @Test
+  fun screenshotTestPrimaryColorSpan_Default() {
+    textView.text = "Test"
```
Comment (tiagoengel) [inline app/src/androidTest/java/net/skyscanner/backpack/text/BpkLinksSpanTests.kt]: What is this test testing? Seems to be texting just the normal text without span, if so we can remove it as the text component already tests this.

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkLinkSpan.kt (hunk 1 above)
Comment (bvitaliyg) [reply Backpack/src/main/java/net/skyscanner/backpack/text/BpkLinkSpan.kt]: Done

File: Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt (hunk 3 above)
Comment (bvitaliyg) [reply Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt]: Done

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkLinksSpanTests.kt (hunk 4 above)
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/text/BpkLinksSpanTests.kt]: Done

**Comment by tiagoengel** [comment]
Looks good to me now, can you update the docs to mention the new link span?

**Pull request merged**